src/document/utils/base.ts

```diff
@@ -226,10 +299,26 @@ export function replayDocument<T, A extends Action, L>(
         };
     }, {} as DocumentOperations<A>);
 
-    // runs all the operations on the new document
+
+    const activeOperationsMap = Object.keys(activeOperations).reduce((acc, curr) => {
+        const scope = curr as keyof DocumentOperations<A>;
+        return {
+            ...acc,
+            [scope]: mapSkippedOperations(activeOperations[scope]),
+        };
+    }, {} as DocumentOperationsIgnoreMap<A>)
+
+    // runs all the operations not ignored on the new document
     // and returns the resulting state
-    const result = sortOperations(activeOperations).reduce(
-        (document, operation) => reducer(document, operation, dispatch),
+    const result = sortMappedOperations(activeOperationsMap).reduce(
+        (document, { ignore, operation }) => {
+            if (ignore) {
+                // ignored operations are replaced by NOOP operations
+                return reducer(document, noop(operation.scope, operation.skip), dispatch);
```

Comment:
Don't we lose the ability to redo skipped operations if we replace them with noop?
Maybe noop should have the original operation as input.

Reply:
in this case, yes.
But the REDO operation is going to be implemented along with the garbage collector https://github.com/powerhouse-inc/document-model/issues/27

this code only process the operations that have been added to the document so it's ok to replace them with `NOOP` operations (if apply).

when you dispatch a new action, and includes skip > 0, the base reducer is going to take care of store skipped operations in the garbage collector history.